## Базовая модель
Линейная регрессия

In [1]:
import sys
from pathlib import Path

project_root = Path("..").resolve()
if str(project_root) not in sys.path:
    sys.path.append(str(project_root))

In [2]:
from src.modeling import run_baseline_pipeline

In [3]:
results = run_baseline_pipeline()
results

{'model': LinearRegression(),
 'model_path': WindowsPath('K:/HSE/hseml-group-project-pe3ricia/models/baseline_linear_regression.pkl'),
 'valid_metrics': {'RMSE': 1433031.9916384702,
  'MAE': 1242797.6562545518,
  'MAPE': np.float64(230.555772890018),
  'R2': -0.0030927931092421534},
 'test_metrics': {'RMSE': 1423656.1892574027,
  'MAE': 1232537.5610119202,
  'MAPE': np.float64(199.33933424737856),
  'R2': -0.0017515808303716351},
 'feature_cols': ['Money Laundering Risk Score',
  'Shell Companies Involved',
  'transaction_year',
  'transaction_month',
  'transaction_dayofweek',
  'transaction_hour',
  'is_illegal',
  'is_reported']}

In [8]:
import pandas as pd

metrics_df = pd.DataFrame(
    [
        {"split": "validation", **results["valid_metrics"]},
        {"split": "test", **results["test_metrics"]},
    ]
)

metrics_df

,split,RMSE,MAE,R2
0,validation,1.433032e+06,1.242798e+06,-0.003093
1,test,1.423656e+06,1.232538e+06,-0.001752


Базовая линейная регрессия на простых числовых признаках показала RMSE около 1.43 млн USD и MAE около 1.24 млн USD на валидационной выборке (и сопоставимые значения на тестовой). Значения R² близки к нулю и слегка отрицательны (около −0.003 на validation и −0.002 на test), что означает, что модель практически не объясняет вариацию таргета и по качеству лишь немногим отличается от наивной стратегии “предсказывать среднее значение суммы для всех транзакций”.

Такой результат соответствует ожиданиям для очень простого baseline’а и подтверждает необходимость более сложных моделей и использования категориальных признаков